src/main/scala/cromwell/binding/WdlNamespace.scala

```diff
@@ -101,6 +102,54 @@ case class NamespaceWithWorkflow(importedAs: Option[String],
     }
   }
 
+  /* Some declarations need a value from the user and some have an expression attached to them.
+   * For the declarations that have an expression attached to it already, evaluate the expression
+   * and return the value for storage in the symbol store
+   */
+  def staticDeclarationsRecursive(userInputs: WorkflowCoercedInputs): Try[WorkflowCoercedInputs] = {
+    import scala.collection.mutable
+    val collected = mutable.Map[String, WdlValue]()
+    class NoFunctions extends WdlFunctions {
+      def getFunction(name: String): WdlFunction = throw new UnsupportedOperationException("No functions should be called in this test")
+    }
+
+    val allDeclarations = workflow.declarations ++ workflow.calls.flatMap {_.task.declarations}
+
+    val evaluatedDeclarations = allDeclarations.filter {_.expression.isDefined}.map {decl =>
```

Comment:
This is an anonymous function which comes complete with its own internal functions. That's a pretty good sign that this is a unit which should be pulled out and then directly called.